src/methods/is.js

```diff
@@ -108,6 +108,16 @@ export function IsCallable(realm: Realm, _func: Value): boolean {
   if (HasCompatibleType(func, FunctionValue)) return true;
   if (func.isSimpleObject()) return false;
 
+  if (func instanceof AbstractObjectValue && !func.values.isTop()) {
+    let result;
+    for (let element of func.values.getElements()) {
+      let isCallable = IsCallable(realm, element);
+      if (result === undefined) result = isCallable;
+      else if (result !== isCallable) func.throwIfNotConcreteObject();
+    }
+    if (result !== undefined) return result;
```

Comment:
This can be an invariant. There is always at least one element if it is not Top.